### PR TITLE
Inline functions redo

### PR DESCRIPTION
**Pull request by hexaredecimal**
I have followed your advice and implemented inline functions from the parser. I also added a function/method for creating hints. I had some time on my hands so I also decide to start working on the run command for qbe. I'm thinking of refactoring the code to a function to reduce the code size of the match expression.

### CODE REVIEW

File: src/command/run.rs [hunk 1]
```diff
@@ -50,6 +51,40 @@ pub fn run(target: Target, in_file: PathBuf) -> Result<(), String> {
...
+            Command::new("gcc")
+                .arg(&asm_path)
+                .arg("-o")
+                .arg(&exe_path)
+                .spawn().unwrap(); 
+
+            // Run the EXE
+            Command::new(exe_path)
+                .spawn().unwrap();
+        }
```
Comment (YerinAlexey) [inline src/command/run.rs:87]: This should be in a different PR as it's not a requirement for the shorthand returns

File: src/parser/parser.rs [hunk 2]
```diff
@@ -139,6 +139,17 @@ impl Parser {
...
+        let new_lines = "\n".repeat(3);
+
+        match &self.raw {
+            Some(_) => format!(
+                "{new_lines}Hint: {}\n",
+                msg,
+            ),
+            None => format!("{new_lines}Hint: {}\n",msg),
+        }
+    }
```
Comment (YerinAlexey) [inline src/parser/parser.rs:152]: This match seems redundant

File: src/parser/rules.rs [hunk 3]
```diff
@@ -159,7 +160,18 @@ impl Parser {
...
+            self.parse_block()?
+        } else if peeked_kind == TokenKind::Assign {
+            self.parse_inline_function()?
+        } else {
+            let token = self.peek()?;
+            let mut error = self.make_error_msg(token.pos, format!("Expected `{{` or `=`, got {}", token.raw));
+            let hint = self.make_hint_msg(format!("Try the following:\nfn {name}(...) = expression\nOr\nfn {name}(...) {{ ... }}"));
+            error.push_str(&hint);
+            return Err(error);
+        };
```
Comment (YerinAlexey) [inline src/parser/rules.rs:174]: I wonder if using a match here instead of else-if would be better

File: src/parser/rules.rs [hunk 4]
```diff
@@ -169,6 +181,15 @@ impl Parser {
         })
     }
 
+    fn parse_inline_function(&mut self) -> Result<Statement, String> {
+        self.next()?; 
+        let expr = self.parse_expression()?;
+        let return_statment = Statement::Return(Some(expr));
+        let statements = vec![return_statment];
+        let scope = vec![];
+        Ok(Statement::Block { statements, scope })
```
Comment (YerinAlexey) [inline src/parser/rules.rs:190]: This could probably use non-shorthand field initialization syntax:

```rust
Ok(Statement::Block {
    statements: vec![return_statement],
    scope: vec![]
})
```

**Review by YerinAlexey — CHANGES_REQUESTED** [review]

File: src/command/run.rs (hunk 1 above)
Comment (garritfra) [reply src/command/run.rs:87]: I personally don't mind the addition here. Saves time and trouble. ✌️

**Comment by garritfra** [comment]
I get an `Error: Expected token` when running this function:

```rs
fn main() {
    println(greet("World"))
}

fn greet(name: string) = "Hello " + name
```

Could you add some tests to the `tests/` directory, and to the parser?

I created a parser test that I used to debug the error:

```rs
#[test]
fn test_parse_inline_function() {
    let raw = "
    fn main() = 1 
    ";
    let tokens = tokenize(raw).unwrap();
    let tree = parse(tokens, Some(raw.to_string()), "".into());
    assert!(tree.is_ok())
}
```

I didn't have enough time to investigate this further though. Trying again later.
